src/Components/test/E2ETest/ServerRenderingTests/FormHandlingTests/FormWithParentBindingContextTest.cs

```diff
@@ -824,19 +807,6 @@ public void FormThatDisappearsBeforeQuiesceDoesNotBind()
         DispatchToFormCore(dispatchToForm);
     }
 
-    [Fact]
-    public void ChangingComponentsToDispatchBeforeQuiesceDoesNotBind()
-    {
-        var dispatchToForm = new DispatchToForm(this)
-        {
-            Url = "forms/switching-components-does-not-bind",
-            FormCssSelector = "form",
-            ExpectedActionValue = null,
-            ShouldCauseInternalServerError = true,
-        };
-        DispatchToFormCore(dispatchToForm);
-    }
```

Comment:
Is there a reason why we are removing this check? The goal here was to ensure that only one component defines a specific named event between the render starts and we go to dispatch the form.

This is meant to avoid circumstances where two components pick the same name and just happen to render at different points in the cycle.

Reply:
I removed it because I thought this previous restriction was just a limitation in the old implementation. The new implementation always knows the complete set of named events and always immediatley untracks any that were removed. 

As such we no longer need to block the form from binding in this case. As long as, by the time we reach quiescence, only one named event is registered with a given name, that's the one to invoke. It's desirable that people should be free to update their component hierarchy arbitrarily. Just because some other form previously existed and used a certain name shouldn't stop components added later from using that name (as long as the old one is now gone).

Reply:
> just happen to render at different points in the cycle.

I think what we care about is the state of the component hierarchy at the point of quiescence. There shouldn't be any random timing issues with this, as quiescence is a deterministic thing.